libensemble/tests/functionality_tests/test_persistent_uniform_gen_decides_stop.py

```diff
@@ -83,7 +83,7 @@
                 sum(counts == init_batch_size) >= ngens
             ), "The initial batch of each gen should be common among initial_batch_size number of points"
             assert (
-                len(np.unique(counts)) > 1
+                len(counts) > 1
```

Comment:
This assert, does not really match the comment. As it is the second "unique" it is really checking that successive generator cycles process diff num of points. To check for diff gen_ended_times - remove the unique (thats what I thought we were doing).

If you want to check after initial batch, you deal with different counts, I'm not sure you can easily ensure that with one gen. That is, if all sims come back, in a given manager cycle, it will be the same.

Reply:
We already check counts[0] == init_batch_size 
I can do
            for i in range(ngens):
                assert counts[i] == init_batch_size, (...
                
would that be sufficient ?